solidity/contracts/converter/types/liquidity-pool-v1/LiquidityPoolV1Converter.sol

```diff
@@ -177,27 +180,31 @@ contract LiquidityPoolV1Converter is LiquidityPoolConverter {
         return amount;
     }
 
-    function _write() internal {
-        uint256 price0 =
-            ISovrynSwapFormula(addressOf(SOVRYNSWAP_FORMULA)).crossReserveTargetAmount(
-                reserveBalance(reserveTokens[0]),
-                reserves[reserveTokens[0]].weight,
-                reserveBalance(reserveTokens[1]),
-                reserves[reserveTokens[1]].weight,
-                1
-            );
-
-        uint256 price1 =
-            ISovrynSwapFormula(addressOf(SOVRYNSWAP_FORMULA)).crossReserveTargetAmount(
-                reserveBalance(reserveTokens[1]),
-                reserves[reserveTokens[1]].weight,
-                reserveBalance(reserveTokens[0]),
-                reserves[reserveTokens[0]].weight,
-                1
-            );
-
-        oracle.write(price0, price1);
-    }
+	function _write() internal {
+		uint256 price0 = ISovrynSwapFormula(addressOf(SOVRYNSWAP_FORMULA)).crossReserveTargetAmount(
+			reserveBalance(reserveTokens[0]),
+			reserves[reserveTokens[0]].weight,
+			reserveBalance(reserveTokens[1]),
+			reserves[reserveTokens[1]].weight,
+			reserveBalance(reserveTokens[0]).div(POOL_PERCENTAGE)
+		);
+
+		uint256 price1 = ISovrynSwapFormula(addressOf(SOVRYNSWAP_FORMULA)).crossReserveTargetAmount(
+			reserveBalance(reserveTokens[1]),
+			reserves[reserveTokens[1]].weight,
+			reserveBalance(reserveTokens[0]),
+			reserves[reserveTokens[0]].weight,
+			reserveBalance(reserveTokens[1]).div(POOL_PERCENTAGE)
+		);
+
+		uint256 decimals0 = reserveTokens[0] == address(etherToken) ? ETHER_DECIMALS : IERC20Token(reserveTokens[0]).decimals();
+		uint256 decimals1 = reserveTokens[1] == address(etherToken) ? ETHER_DECIMALS : IERC20Token(reserveTokens[1]).decimals();
+
+		price0 = price0.mul(10**decimals0).div(reserveBalance(reserveTokens[0]).div(POOL_PERCENTAGE));
+		price1 = price1.mul(10**decimals1).div(reserveBalance(reserveTokens[1]).div(POOL_PERCENTAGE));

```

Comment:
please avoid calculating the swap size twice (here and as argument on the formula call), but store it in a local variable first to save gas costs